ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoiceNotifier.java

```diff
@@ -90,8 +93,31 @@ public void onAttestationsDue(final UInt64 slot) {
     eventThread.execute(() -> internalAttestationsDue(slot));
   }
 
-  public SafeFuture<Optional<Bytes8>> getPayloadId() {
-    return eventThread.execute(() -> lastPayloadId);
+  public SafeFuture<Optional<Bytes8>> getPayloadId(Bytes32 beaconBlockRoot, UInt64 slot) {
+    return eventThread.execute(
+        () -> {
+          if (lastPayloadId.isPresent()) {
```

Comment:
We tend to prefer exiting early so would normally write:
```
if (lastPayloadId.isEmpty()) {
   LOG...
   return;
}
```
Just saves the extra indentation mostly.